### PR TITLE
Feature/iexec upgrade

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
# upgrade to iexec@8.10.0

- support for ethers AbstractProvider and AbstractSigner

# misc

- doc regenerated (it seems this was forgotten for a while, sorry for that)

### CODE REVIEW

File: src/oracleFactory/OracleFactory.ts [hunk 1]
```diff
@@ -53,7 +58,11 @@ class IExecOracleFactory {
    * @param options Optional configuration options OracleFactory.
    */
   constructor(
-    ethProvider: Eip1193Provider | Web3SignerProvider,
+    ethProvider:
+      | Eip1193Provider
+      | Web3SignerProvider
+      | AbstractSigner
+      | BrowserProvider,
```
Comment (cedric25) [inline src/oracleFactory/OracleFactory.ts:65]: In dataprotector-sdk and web3mail-sdk we have:
```
    ethProvider?:
      | AbstractProvider
      | AbstractSigner
      | Eip1193Provider
      | Web3SignerProvider
      | string,
```

Should we add AbstractProvider here in oracle-factory?

Should we also add BrowserProvider in dataprotector-sdk and web3mail-sdk?

File: src/oracleFactory/OracleReader.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ class IExecOracleReader {
     ethProviderOrNetwork:
       | Web3ReadOnlyProvider
       | Web3SignerProvider
+      | AbstractProvider
```
Comment (abbesBenayache) [inline src/oracleFactory/OracleReader.ts:54]:
```suggestion
      | AbstractProvider
      | BrowserProvider,
```

**Review by cedric25 — APPROVED** [review]

**Review by abbesBenayache — APPROVED** [review]
LGTM!

File: src/oracleFactory/OracleReader.ts (hunk 2 above)
Comment (PierreJeanjacquot) [reply src/oracleFactory/OracleReader.ts:54]: no need

BrowserProvider extends JsonRpcApiPollingProvider
JsonRpcApiPollingProvider extends JsonRpcApiProvider
JsonRpcApiProvider extends AbstractProvider

**Pull request merged**